File: Api/modules/Posts/ProfesiNet.Posts.Core/DAL/Entities/Post.cs
namespace ProfesiNet.Posts.Core.DAL.Entities;

public class Post
{
    public Guid Id { get; set; }
    
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public string? ImageId { get; set; }
    public DateTime PublishedAt { get; set; }
    
    public virtual ICollection<Share>? Shares { get; set; } = new List<Share>();
    public virtual ICollection<Comment>? Comments { get; set; }
    public virtual ICollection<PostLike> Likes { get; set; } = new List<PostLike>();
    public Guid? CreatorId { get; set; }
    public virtual Creator? Creator { get; set; } = null!;

}

File: Client/ProfesiNetClient/src/app/modules/interfaces/Post.ts
export interface Post {
    id: string
    creatorId: string
    creatorName: string
    creatorSurname: string
    creatorProfilePicture: string
    imageUrl: string
    description: string
    publishedAt: string
    likesCount: number
    commentsCount: number
    sharesCount: number
    isLiked: boolean
    isShared: boolean
    
    shares: Shares[]
    likes: Likes[]
}

export interface Shares {
    id: string
    creatorId: string
    postId: string
}


export interface Likes {
    id: string
    creatorId: string
    postId: string
}
